### ISSUE TITLE
Compile 64bits version with vc11 : Wow64GetThreadContext problem

### ISSUE THREAD

**Issue by ryzom-pipeline** [ryzom/ryzomcore#50]
**[Original report](https://bitbucket.org/ryzom/ryzomcore/issue/50) by S (Bitbucket: [S](https://bitbucket.org/S), GitHub: [S](https://github.com/S)).**

----------------------------------------

**ENV :**
On Win7
With VS 2012 Express
With WinSDK 8
In 64 bits 

**ERROR :** 
while compiling Wow64GetThreadContext not found
ryzomcore\code\nel\src\misc\mem_displayer.cpp

**CAUSE : **

* mem_displayer.cpp use Wow64GetThreadContext 

```
#!c++

#ifdef NL_OS_WIN64
	if (Wow64GetThreadContext (GetCurrentThread(), &context) == FALSE)
#else
	if (GetThreadContext (GetCurrentThread(), &context) == FALSE)
#endif
```

* WinBase.h in WinSDK8 define minimal version for Wow64GetThreadContext 
(previous version of WinSDK do not define minimal version)

```
#!c++

#if (_WIN32_WINNT >= 0x0600)

WINBASEAPI
BOOL
WINAPI
Wow64GetThreadContext(
    _In_    HANDLE hThread,
    _Inout_ PWOW64_CONTEXT lpContext
    );

WINBASEAPI
BOOL
WINAPI
Wow64SetThreadContext(
    _In_ HANDLE hThread,
    _In_ CONST WOW64_CONTEXT *lpContext
    );

#endif // (_WIN32_WINNT >= 0x0600)
```

** QUICK SOLUTION **

Win2000 do not exist in 64 bits.

WinXP 64Bits do not define Wow64GetThreadContext  (see https://portfolio.assembla.com/spaces/winappdbg/tickets/26#/activity/ticket:)

Change 
ryzomcore\code\nel\include\nel\misc\types_nl.h

```
#!c++

#	ifdef _WIN64
#		define NL_OS_WIN64
#		define _WIN32_WINNT 0x0600  // force VISTA minimal version in 64 bits
#		ifndef NL_NO_ASM		
#			define NL_NO_ASM
#		endif
#	endif
```

Note : 
WIN32_WINNT 0x0500 is for Win2000
WIN32_WINNT 0x0501 is for WinXP
WIN32_WINNT 0x0600 is for VISTA

**Comment by ryzom-pipeline** [comment]
**Original comment by Cédric Ochs (Bitbucket: [Cédric OCHS](https://bitbucket.org/Cédric OCHS), ).**

----------------------------------------

You're right, it must be a bug in Windows SDK 7.1 where Wow64GetThreadContext is defined even for Windows 2000...

I'll commit your fix, thanks a lot :)

**Comment by ryzom-pipeline** [comment]
**Original comment by Cédric Ochs (Bitbucket: [Cédric OCHS](https://bitbucket.org/Cédric OCHS), ).**

----------------------------------------

Fixed in e0764d2

Sorry I forgot the put the issue id in commit description :(
